### PR TITLE
Fix flyTo rectangle in 2D

### PR DESCRIPTION
**Pull request by novacto2**
This is for #3688 

Zoom limits check leads to a change of coordinates.

### CODE REVIEW

**Comment by hpinkos** [comment]
Thanks @novacto2!  Can you please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) so we can review this?

**Comment by pjcozzi** [comment]
We now have a CLA from @novacto2.  Thanks!

**Comment by hpinkos** [comment]
I'm not sure if deleting that code is the correct solution here.
@bagnell, can you look at this?

**Comment by novacto2** [comment]
I've made some changes @hpinkos

It seems that the transformation of the camera position from cartesian to cartographic gives a negative height (in this specific case - rectangle in 2D). The computed height is stored in the 'z' component of cartesian, so the zoom limits should be tested on this number.

**Comment by hpinkos** [comment]
Thanks for the update!
I can confirm this fixes the issue.  And all of the tests pass.
We'll probably want to add a unit test for this.  @novacto2 would you mind doing that? See our [testing guide](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Documentation/Contributors/TestingGuide/README.md) for information

@bagnell, does this code look good to you?

File: Source/Scene/Camera.js [hunk 1]
```diff
@@ -2558,12 +2558,16 @@ define([
 
         // Make sure camera doesn't zoom outside set limits
         if (defined(sscc)) {
-            destinationCartographic.height = CesiumMath.clamp(destinationCartographic.height, sscc.minimumZoomDistance, sscc.maximumZoomDistance);
-
-            //Only change if we clamped the height
-            if (destinationCartographic.height === sscc.minimumZoomDistance || destinationCartographic.height === sscc.maximumZoomDistance) {
-                destination = ellipsoid.cartographicToCartesian(destinationCartographic);
-            }
+                if (mode === SceneMode.SCENE2D && isRectangle) {
```
Comment (bagnell) [inline Source/Scene/Camera.js]: This is true for Columbus view as well. I would change this to `mode !== SceneMode.SCENE3D`.

**Comment by bagnell** [comment]
Looks good besides the one comment and formatting. @hpinkos Can you merge when you're ready?

**Comment by novacto2** [comment]
@bagnell you're right, thanks:) I changed the condition and formatting.

@hpinkos unit tests were added

**Comment by hpinkos** [comment]
Alright, looks great @novacto2! Thanks!
Can you merge master?  It looks like there's a conflict.
Also, add a line to [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) for this bug fix, then we'll get this merged in.

**Comment by novacto2** [comment]
Is it ok @hpinkos ?

**Comment by hpinkos** [comment]
Thanks @novacto2!  Just a few tiny things:
Please add yourself to [CONTRIBUTORS.md]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTORS.md)

And it looks like your formatting is still off.  Please see the [Coding Guide](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#formatting)
If you are coding in (Eclipse)[https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/EclipseGuide] or (WebStorm)[https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/WebStormGuide], highlighting your changes and running the autoformatter should fix the problem.

**Comment by novacto2** [comment]
I'm sorry about the mistakes. Hope it's ok now.

Thanks for going easy on me @hpinkos :)

**Comment by hpinkos** [comment]
Looks great!  Thanks @novacto2!

**Pull request merged**